### PR TITLE
refactor SimpleHellaCacheIF to chisel3.

### PR DESCRIPTION
**Pull request by SingularityKChen**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

Refactor SimpleHellaCacheIF.scala to chisel3 style.

### CODE REVIEW

File: src/main/scala/rocket/SimpleHellaCacheIF.scala [hunk 1]
```diff
@@ -17,20 +18,20 @@ import freechips.rocketchip.util._
...
+  })
 
   // Registers to store the sent request
   // When a request is sent the first time,
   // it is stored in one of the reqs registers
   // and the corresponding inflight bit is set.
   // The reqs register will be deallocated once the request is
   // successfully completed.
-  val inflight = Reg(init = UInt(0, depth))
+  val inflight = RegInit(init = 0.U(depth.W))
```
Comment (sequencer) [inline src/main/scala/rocket/SimpleHellaCacheIF.scala:34]:
```suggestion
  val inflight = RegInit(0.U(depth.W))
```

File: src/main/scala/rocket/SimpleHellaCacheIF.scala [hunk 2]
```diff
@@ -40,8 +41,8 @@ class SimpleHellaCacheIFReplayQueue(depth: Int)
   // successfully completed, at which time the request is dequeued.
   // No new requests will be made or other replays attempted until the head
   // of the nackq is successfully completed.
-  val nackq = Module(new Queue(UInt(width = log2Up(depth)), depth))
-  val replaying = Reg(init = Bool(false))
+  val nackq = Module(new Queue(UInt(log2Up(depth).W), depth))
+  val replaying = RegInit(init = false.B)
```
Comment (sequencer) [inline src/main/scala/rocket/SimpleHellaCacheIF.scala:45]:
```suggestion
  val replaying = RegInit(false.B)
```

File: src/main/scala/rocket/SimpleHellaCacheIF.scala [hunk 3]
```diff
@@ -131,6 +132,7 @@ class SimpleHellaCacheIF(implicit p: Parameters) extends Module
   replayq.io.nack.bits := s2_req_tag
   replayq.io.resp := io.cache.resp
   io.requestor.resp := io.cache.resp
+  io <> DontCare
```
Comment (sequencer) [inline src/main/scala/rocket/SimpleHellaCacheIF.scala:135]: I still have no idea why porting to chisel3 has different connection semantic.

Comment (SingularityKChen) [reply src/main/scala/rocket/SimpleHellaCacheIF.scala]: There are some IOs that are not connected which is found in CI test 7 (log is shown below). In Chisel 2, the compile options is `chisel3.core.ExplicitCompileOptions.NotStrict` and whether it means the IO initialization is not checked before?

```text
Exception in thread "main" firrtl.passes.PassExceptions: 
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.canAcceptStoreThenRMW <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.tlbMiss <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.ae.ld <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.blocked <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_nack <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_gpa <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.cache.keep_clock_enabled <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.clock_enabled <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.release <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.storeBufferEmptyAfterStore <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.gf.ld <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.ma.ld <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.ae.st <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.gf.st <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_nack_cause_raw <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.ma.st <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.cache.s1_data.mask <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.pf.ld <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.grant <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.canAcceptStoreThenLoad <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_uncached <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_xcpt.pf.st <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.storeBufferEmptyAfterLoad <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.ordered <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.acquire <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_gpa_is_pte <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.perf.canAcceptLoadThenLoad <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.replay_next <= VOID
firrtl.passes.CheckInitialization$RefNotInitializedException:  : [module SimpleHellaCacheIF]  Reference io is not fully initialized.
   : io.requestor.s2_paddr <= VOID
firrtl.passes.PassException: 29 errors detected!
```

**Review by sequencer — APPROVED** [review]

**Review by sequencer — APPROVED** [review]

**Pull request merged**